apps/web/src/pages/api/onramp-url-sign/generate-mercuryo-sig.ts

```diff
@@ -0,0 +1,28 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import crypto from 'crypto'
+import { z } from 'zod'
+
+const payloadSchema = z.object({
+  walletAddress: z.string(),
+})
+
+export default function handler(req: NextApiRequest, res: NextApiResponse): void {
+  try {
+    const payload = req.body
+    if (!payload) {
+      throw new Error('Payload is required.')
+    }
+    const validPayload = payloadSchema.safeParse(payload)
+    if (!validPayload.success) {
+      throw new Error('payload has the incorrect shape. please check you types')
+    }
+
+    const { walletAddress } = validPayload.data
+
+    const hash = crypto.createHash('sha512').update(`${walletAddress}${process.env.MERCURYO_SECRET_KEY}`).digest('hex')
```

Comment:
add guard when no `MERCURYO_SECRET_KEY` env

Reply:
can you try using edge function here?


```js
export const config = {
  runtime: 'edge',
}

// function hander...
const encoder = new TextEncoder();
const data = encoder.encode(`${walletAddress}${process.env.MERCURYO_SECRET_KEY}`);
const hashBuffer = await crypto.subtle.digest('SHA-512', data);
const hashArray = Array.from(new Uint8Array(hashBuffer));
const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('');
```

Reply:
resolved moved this to BE